### PR TITLE
Add more Java versions to Travis CI build matrix

### PR DESCRIPTION
**Pull request by joschi**
In order to cover the Java/JDK versions we want to support with Dropwizard with Travis CI, this pull request adds OpenJDK 8 through 11 (EA) to the Travis CI build matrix.

There were some minor changes in the Maven build, mostly adding build profiles for the respective Java/JDK versions.

The following issues are open until Java 11 is completely supported without hacks:

* https://github.com/jboss-javassist/javassist/pull/198
* https://liquibase.jira.com/browse/CORE-3262 (this also affects Java 10)

### CODE REVIEW

**Comment by jplock** [comment]
@nickbabcock @joschi is this an issue with the SelfValidatorTest or javaassist?  It only seems to be failing on jdk8 for some reason.

```
ERROR] complexExample(io.dropwizard.validation.SelfValidationTest)  Time elapsed: 0.253 s  <<< ERROR!
java.lang.NoClassDefFoundError: java/lang/StackWalker$Option
	at io.dropwizard.validation.SelfValidationTest.complexExample(SelfValidationTest.java:198)
Caused by: java.lang.ClassNotFoundException: java.lang.StackWalker$Option
	at io.dropwizard.validation.SelfValidationTest.complexExample(SelfValidationTest.java:198)
[ERROR] invalidExample(io.dropwizard.validation.SelfValidationTest)  Time elapsed: 0.005 s  <<< ERROR!
java.lang.NoClassDefFoundError: Could not initialize class javassist.util.proxy.DefineClassHelper
	at io.dropwizard.validation.SelfValidationTest.invalidExample(SelfValidationTest.java:210)
[ERROR] subClassExample(io.dropwizard.validation.SelfValidationTest)  Time elapsed: 0.006 s  <<< ERROR!
java.lang.NoClassDefFoundError: Could not initialize class javassist.util.proxy.DefineClassHelper
	at io.dropwizard.validation.SelfValidationTest.subClassExample(SelfValidationTest.java:142)
[ERROR] annotatedSubClassExample(io.dropwizard.validation.SelfValidationTest)  Time elapsed: 0.005 s  <<< ERROR!
java.lang.NoClassDefFoundError: Could not initialize class javassist.util.proxy.DefineClassHelper
	at io.dropwizard.validation.SelfValidationTest.annotatedSubClassExample(SelfValidationTest.java:153)
[ERROR] overridingSubClassExample(io.dropwizard.validation.SelfValidationTest)  Time elapsed: 0.005 s  <<< ERROR!
java.lang.NoClassDefFoundError: Could not initialize class javassist.util.proxy.DefineClassHelper
	at io.dropwizard.validation.SelfValidationTest.overridingSubClassExample(SelfValidationTest.java:164)
[ERROR] testDirectContextUsage(io.dropwizard.validation.SelfValidationTest)  Time elapsed: 0.004 s  <<< ERROR!
java.lang.NoClassDefFoundError: Could not initialize class javassist.util.proxy.DefineClassHelper
	at io.dropwizard.validation.SelfValidationTest.testDirectContextUsage(SelfValidationTest.java:190)
[ERROR] failingExample(io.dropwizard.validation.SelfValidationTest)  Time elapsed: 0.004 s  <<< ERROR!
java.lang.NoClassDefFoundError: Could not initialize class javassist.util.proxy.DefineClassHelper
	at io.dropwizard.validation.SelfValidationTest.failingExample(SelfValidationTest.java:134)
[ERROR] multipleTestingOfSameClass(io.dropwizard.validation.SelfValidationTest)  Time elapsed: 0.003 s  <<< ERROR!
java.lang.NoClassDefFoundError: Could not initialize class javassist.util.proxy.DefineClassHelper
	at io.dropwizard.validation.SelfValidationTest.multipleTestingOfSameClass(SelfValidationTest.java:180)
[ERROR] correctExample(io.dropwizard.validation.SelfValidationTest)  Time elapsed: 0.003 s  <<< ERROR!
java.lang.NoClassDefFoundError: Could not initialize class javassist.util.proxy.DefineClassHelper
	at io.dropwizard.validation.SelfValidationTest.correctExample(SelfValidationTest.java:172)
```

**Comment by joschi** [comment]
@jplock Javassist 3.23.0-GA doesn't support Java 8 anymore and Javassist 3.22.0-GA doesn't support Java 11.

We'll either have to stop supporting Java 8 (EOL soon), not support Java 11 (released soon), or rewrite the functionality with a different library (e. g. Byte Buddy).

I think I'll give Byte Buddy a try if I have a few minutes to spare.

**Comment by jplock** [comment]
@joschi maybe when the next release comes out containing https://github.com/jboss-javassist/javassist/pull/204, the tests will work again on JDK8.  That puts us in a tricky position.  We should support the latest LTS release (JDK11), but I'm not sure we should lose JDK8 support in DW 2.0.0 or not yet.

File: dropwizard-bom/pom.xml [hunk 1]
```diff
@@ -88,6 +88,18 @@
...
+                <artifactId>jaxb-api</artifactId>
+                <version>2.3.0</version>
+                <scope>runtime</scope>
+            </dependency>
+            <dependency>
+                <groupId>javax.activation</groupId>
+                <artifactId>javax.activation-api</artifactId>
+                <version>1.2.0</version>
+                <scope>runtime</scope>
+            </dependency>
```
Comment (RohanNagar) [inline dropwizard-bom/pom.xml]: Does adding the two dependencies above mean that Dropwizard applications will no longer need `--add-modules java.xml.bind` on Java 9+? Right now I need to manually include that option when running on Java 9

Comment (joschi) [reply dropwizard-bom/pom.xml]: @RohanNagar Yes, the additional command line parameters for java.xml.bind can be removed.

**Comment by joschi** [comment]
The pull request is done and ready to merge from my perspective.

The AppVeyor build failed for unrelated reasons (address already in use in parallel tests).

**Pull request merged**